tests/performance/event_generator/tests/test_syscheckeventgenerator.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (C) 2024, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+"""Unit tests for the SyscheckEventGenerator class."""
+
+import os
+from pathlib import Path
+
+import pytest
+
+from event_generator import SyscheckEventGenerator
+
+
+@pytest.fixture
+def setup_syscheck_generator(tmp_path: Path) -> tuple:
+    """Setup for SyscheckEventGenerator with a temporary directory.
+
+    Args:
+        tmp_path (LocalPath): Temporary directory provided by pytest.
+
+    Returns:
+        tuple: A tuple containing an instance of SyscheckEventGenerator and the path to the directory.
+    """
+    path = tmp_path
+    generator = SyscheckEventGenerator(rate=1, path=str(path), operations=5)
+    return generator, path
+
+
+def test_file_operations(setup_syscheck_generator: tuple):
+    """Test file operations (create, modify, delete) performed by SyscheckEventGenerator.
+
+    Args:
+        setup_syscheck_generator (fixture): Fixture that provides a SyscheckEventGenerator and a path.
+    """
+    generator, path = setup_syscheck_generator
+    generator.start()
+    # Since operations are random, we check if any file exists or not after operations
+    assert any(path.iterdir()
+               ), "There should be some file operations in the directory"
```

Comment:
This test is too generic, we can determinate if it's working as expected through this method. We should try another approach

Reply:
Fixed here: https://github.com/wazuh/wazuh-qa/pull/5697/commits/c4e478c25a34385c2e9139c06a9d5d20f2b753a7